webapp/src/router.tsx

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useEffect, useMemo} from 'react'
+import {
+    Router,
+    Redirect,
+    Switch,
+    useRouteMatch,
+    useHistory,
+    generatePath,
+} from 'react-router-dom'
+
+import {createBrowserHistory} from 'history'
+
+import {IAppWindow} from './types'
+import BoardPage from './pages/boardPage'
+import ChangePasswordPage from './pages/changePasswordPage'
+import WelcomePage from './pages/welcome/welcomePage'
+import ErrorPage from './pages/errorPage'
+import LoginPage from './pages/loginPage'
+import RegisterPage from './pages/registerPage'
+import {Utils} from './utils'
+import octoClient from './octoClient'
+import {setGlobalError, getGlobalError} from './store/globalError'
+import {useAppSelector, useAppDispatch} from './store/hooks'
+import {UserSettings} from './userSettings'
+import FBRoute from './route'
+
+declare let window: IAppWindow
+
+const UUID_REGEX = new RegExp(/^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$/)
+
+function HomeToCurrentTeam(props: {path: string, exact: boolean}) {
+    return (
+        <FBRoute
+            path={props.path}
+            exact={props.exact}
+            loginRequired={true}
+            render={() => {
+                const teamID = (window.getCurrentTeamId && window.getCurrentTeamId()) || ''
+
+                if (UserSettings.lastBoardId) {
+                    const lastBoardID = UserSettings.lastBoardId[teamID]
+                    const lastViewID = UserSettings.lastViewId[lastBoardID]
+
+                    return <Redirect to={`/team/${teamID}/${lastBoardID}/${lastViewID}`}/>
+                }
+
+                return <Redirect to={`/team/${teamID}`}/>
+            }}
+        />
+    )
+}
+
+function WorkspaceToTeamRedirect() {
+    const match = useRouteMatch<{boardId: string, viewId: string, cardId?: string, workspaceId?: string}>()
+    const history = useHistory()
+    useEffect(() => {
+        octoClient.getBoard(match.params.boardId).then((board) => {
+            if (board) {
+                history.replace(generatePath('/team/:teamId/:boardId?/:viewId?/:cardId?', {
+                    teamId: board?.teamId,
+                    boardId: board?.id,
+                    viewId: match.params.viewId,
+                    cardId: match.params.cardId,
+                }))
+            }
+        })
+    }, [])
+    return null
+}
+
+function GlobalErrorRedirect() {
```

Comment:
In transformed the global error redirect to a component that encapsulates that behavior.